SGoopas/Assets/Scripts/Game/3D/ObjectInterract/ObjInteractableBase.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+/*
+ * Contains an implementation for highlighting objects and an interface for how the player uses the interactables.
+ */
+public abstract class ObjInteractableBase : MonoBehaviour
+{
+    // This allows you to configure multi-object interactables.
+    public GameObject[] associatedObjects;
+
+    private static string highlightName = "HighlightMaterial";
+    private static Material highlightMaterial;
+    private static List<Material> originalMaterials;
+
+    static Material GetHighlightMaterial()
+    {
+        if (!highlightMaterial) {
+            Shader highlightShader = Shader.Find("Outline/Transparent");
+            Assert.IsNotNull(highlightShader, "Could not load highlight shader");
+            highlightMaterial = new Material(highlightShader);
+            highlightMaterial.name = highlightName;
+        }
+        return highlightMaterial;
+    }
+
+    public virtual void Awake()
+    {
+        // If the associatedObjects field isn't filled, assume the interactable only applies to the root object.
+        if (associatedObjects == null || associatedObjects.Length == 0)
+        {
+            associatedObjects = new GameObject[] { gameObject };
+        }
+    }
+
+    public virtual void HighlightObject() {
+        originalMaterials = new List<Material>();
+        foreach (GameObject highlightObj in associatedObjects)
+        {
+            Material ogMaterial = highlightObj.GetComponent<Renderer>().material;
+            Material[] highlightMaterialSet = {ogMaterial, GetHighlightMaterial()};
+            highlightObj.GetComponent<Renderer>().materials = highlightMaterialSet;
```

Comment:
I didnt know 'highlightObj.GetComponent<Renderer>().materials' could be fed an array of materials. what would this do then?

Reply:
It just applies both shaders, one after the other. Try it.

Reply:
I see. interesting, I did not know that lol. Thanks!